test/integration/protocol_integration_test.cc

```diff
@@ -4769,7 +4769,8 @@ class AllowForceFail : public Api::OsSysCallsImpl {
   bool fail_ = false;
 };
 
-TEST_P(ProtocolIntegrationTest, HandleUpstreamSocketCreationFail) {
+TEST_P(ProtocolIntegrationDeathTest, HandleUpstreamSocketCreationFail) {
+  GTEST_FLAG_SET(death_test_style, "threadsafe");
```

Comment:
Why is this needed here (and in some of the other tests)? I thought it is set once in the test_runner.

Reply:
I did not go digging into why. All I can tell you is that anywhere I set it explicitly, the tests failed without it.